sourmash/__init__.py

```diff
@@ -20,7 +20,9 @@
 from . import sbt_storage
 from . import signature
 
-# retrieve VERSION from sourmash/VERSION.
-thisdir = os.path.dirname(__file__)
-version_file = open(os.path.join(thisdir, 'VERSION'))
-VERSION = version_file.read().strip()
+from pkg_resources import get_distribution, DistributionNotFound
+try:
+    VERSION = get_distribution(__name__).version
+except DistributionNotFound:
+    # package is not installed
+    pass
```

Comment:
so, um, what happens if DistributionNotFound - what is VERSION then?

Reply:
Good point. I would say it's better to `raise SystemExit()` with a message saying that the installation has issues (either because it was not installed properly or at all...)

Reply:
This is 'fixed' on the newest commit: `sourmash/version.py` is written when the package is generated (as sdist or wheel), but we don't commit it to the repository.

It still doesn't work with github archives (because there is not enough info), but it gives a recomendadion at the end pointing to the right command:
```bash
$ pip install https://github.com/dib-lab/sourmash/archive/feat/scm_version.tar.gz 
Looking in links: file:///home/luizirber/.cache/pip/wheelhouse
Collecting https://github.com/dib-lab/sourmash/archive/feat/scm_version.tar.gz
  Downloading https://github.com/dib-lab/sourmash/archive/feat/scm_version.tar.gz
     | 14.4MB 81.6MB/s
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-req-build-xs11eam_/setup.py", line 83, in <module>
        setup(**SETUP_METADATA)
      File "/home/luizirber/Envs/sourtest-dK1vLKWi/lib/python3.7/site-packages/setuptools/__init__.py", line 140, in setup
        return distutils.core.setup(**attrs)
      File "/usr/lib64/python3.7/distutils/core.py", line 108, in setup
        _setup_distribution = dist = klass(attrs)
      File "/home/luizirber/Envs/sourtest-dK1vLKWi/lib/python3.7/site-packages/setuptools/dist.py", line 370, in __init__
        k: v for k, v in attrs.items()
      File "/usr/lib64/python3.7/distutils/dist.py", line 292, in __init__
        self.finalize_options()
      File "/home/luizirber/Envs/sourtest-dK1vLKWi/lib/python3.7/site-packages/setuptools/dist.py", line 529, in finalize_options
        ep.load()(self, ep.name, value)
      File "/tmp/pip-req-build-xs11eam_/.eggs/setuptools_scm-3.1.0-py3.7.egg/setuptools_scm/integration.py", line 23, in version_keyword
        dist.metadata.version = get_version(**value)
      File "/tmp/pip-req-build-xs11eam_/.eggs/setuptools_scm-3.1.0-py3.7.egg/setuptools_scm/__init__.py", line 142, in get_version
        parsed_version = _do_parse(config)
      File "/tmp/pip-req-build-xs11eam_/.eggs/setuptools_scm-3.1.0-py3.7.egg/setuptools_scm/__init__.py", line 111, in _do_parse
        "use git+https://github.com/user/proj.git#egg=proj" % config.absolute_root
    LookupError: setuptools-scm was unable to detect version for '/tmp/pip-req-build-xs11eam_'.
    
    Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.
    
    For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-req-build-xs11eam_/
```

And, after running the command, it gives back the right version:
```bash
$ pip install git+https://github.com/dib-lab/sourmash@feat/scm_version#egg=sourmash
Looking in links: file:///home/luizirber/.cache/pip/wheelhouse
Collecting sourmash from git+https://github.com/dib-lab/sourmash@feat/scm_version#egg=sourmash
  Cloning https://github.com/dib-lab/sourmash (to revision feat/scm_version) to /tmp/pip-install-wk3kd7rg/sourmash
Collecting screed>=0.9 (from sourmash)
Collecting ijson (from sourmash)
  Using cached https://files.pythonhosted.org/packages/7f/e9/8508c5f4987ba238a2b169e582c1f70a47272b22a2f1fb06b9318201bb9e/ijson-2.3-py2.py3-none-any.whl
Collecting khmer>=2.1 (from sourmash)
  Using cached https://files.pythonhosted.org/packages/3f/1e/931281ac388cbb15d40319d0dcd54cfc3f1d972da5de2fa3063facedc966/khmer-2.1.1-cp37-cp37m-manylinux1_x86_64.whl
Collecting bz2file (from screed>=0.9->sourmash)
Building wheels for collected packages: sourmash
  Running setup.py bdist_wheel for sourmash ... done
  Stored in directory: /tmp/pip-ephem-wheel-cache-q794qi7c/wheels/f3/23/71/e3071a4b240d19fa3f2023e96808ffece11f5d43a09c6bac0f
Successfully built sourmash
Installing collected packages: bz2file, screed, ijson, khmer, sourmash
Successfully installed bz2file-0.98 ijson-2.3 khmer-2.1.1 screed-1.0 sourmash-2.0.0a10.dev6+g04948ad

$ python
Python 3.7.0 (default, Sep 15 2018, 19:13:07) 
[GCC 8.2.1 20180831] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sourmash
>>> sourmash.VERSION
'2.0.0a10.dev6+g04948ad'
```